protocol/chainlib/common_test_utils.go

```diff
@@ -141,9 +154,23 @@ func CreateChainLibMocks(ctx context.Context, specIndex string, apiInterface str
 			return nil, nil, nil, closeServer, nil, err
 		}
 	} else {
-		mockServer := httptest.NewServer(serverCallback)
-		closeServer = mockServer.Close
-		endpoint.NodeUrls = append(endpoint.NodeUrls, common.NodeUrl{Url: mockServer.URL, Addons: addons})
+		var mockWebSocketServer *httptest.Server
+		var wsUrl string
+		if wsServerCallback != nil {
+			mockWebSocketServer = httptest.NewServer(wsServerCallback)
+			wsUrl = "ws" + strings.TrimPrefix(mockWebSocketServer.URL, "http")
+		}
+		mockHttpServer := httptest.NewServer(httpServerCallback)
+		closeServer = func() {
+			mockHttpServer.Close()
+			if wsServerCallback != nil {
+				mockWebSocketServer.Close()
+			}
+		}
+		endpoint.NodeUrls = append(endpoint.NodeUrls, common.NodeUrl{Url: mockHttpServer.URL, Addons: addons})
+		if wsServerCallback != nil {
```

Comment:
if we dont add websocket listener most tests will fail due to subscription in spec but no ws provided.

Reply:
solved here: https://github.com/lavanet/lava/pull/1462/commits/57c0d8b206465664f39cf74b6c228a10f7ae9632